Items/Equipable/MagnetismRing.cs

```diff
@@ -31,19 +27,30 @@ public override void SetDefaults()
 		
 		public override void UpdateAccessory(Player player, bool hideVisual)
 		{
-			player.GetModPlayer<MiscPlayer>().Magnet = true;
+			player.treasureMagnet = true;
 			player.goldRing = true;
 			player.manaMagnet = true;
 		}
 		
 		public override void AddRecipes()
 		{
-			CreateRecipe()
-				.AddIngredient(ModContent.ItemType<UniversalMagnet>())
-				.AddIngredient(ItemID.CelestialMagnet)
-				.AddIngredient(ItemID.GoldRing)
-				.AddTile(TileID.TinkerersWorkbench)
-				.Register();
+			if (ModContent.GetInstance<ServerConfig>().Magnet)
+			{
+				CreateRecipe()
+					.AddIngredient(ModContent.ItemType<UniversalMagnet>())
+					.AddIngredient(ItemID.CelestialMagnet)
+					.AddIngredient(ItemID.GoldRing)
+					.AddTile(TileID.TinkerersWorkbench)
+					.Register();
+
+				CreateRecipe()
+						.AddIngredient(ItemID.TreasureMagnet)
+						.AddIngredient(ItemID.CelestialMagnet)
+						.AddIngredient(ItemID.GoldRing)
+						.AddTile(TileID.TinkerersWorkbench)
+						.Register();
+			}
```

Comment:
This recipe shouldn't depend on the Universal Magnet config, probably

Reply:
But it would point to a unloaded item?

Reply:
Well we separated the Universal Magnet config and the Magnetism Ring config. It should be possible to craft the Magnetism Ring without the Universal Magnet by using the vanilla Treasure Magnet instead.

Reply:
Yep your right. I wasn't looking hard enough at the name of the file.
I came back from a trip after working on all of this was originally hoping to submit this pr before.